File: Deepseek-Coder-33b-Instruct/stage2/routine_directive.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

#pragma acc routine seq
void vector_add(int* a, int* b, int* c, int n) {
    for(int i = 0; i < n; i++) {
        c[i] = a[i] + b[i];
    }
}

int main() {
    const int n = 1000000;
    int a[n], b[n], c[n];

    srand(time(0));
    for(int i = 0; i < n; i++) {
        a[i] = rand() % 100;
        b[i] = rand() % 100;
    }

    #pragma acc data copyin(a[0:n], b[0:n]) copyout(c[0:n])
    {
        #pragma acc parallel loop
        for(int i = 0; i < n; i++) {
            vector_add(a, b, c, n);
        }
    }

    for(int i = 0; i < n; i++) {
        if(c[i] != a[i] + b[i]) {
            std::cout << "Error at index " << i << std::endl;
            return 1;
        }
    }

    std::cout << "Success" << std::endl;
    return 0;
}

File: Deepseek-Coder-33b-Instruct/stage2/routine_directive.f90
!T1:routine directive,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 1000
        REAL, DIMENSION(N) :: a, b, c
        INTEGER :: i

        ! Initialize arrays
        DO i = 1, N
          a(i) = i * 1.0
          b(i) = i * 2.0
        END DO

        !$acc data copyin(a, b) copyout(c)
        !$acc parallel loop
        DO i = 1, N
          c(i) = a(i) + b(i)
        END DO
        !$acc end data

        ! Check the results
        DO i = 1, N
          IF (c(i) .ne. a(i) + b(i)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
